### PR TITLE
Brayns 300 replace free image

### PR DESCRIPTION
**Pull request by Adrien4193**
- Image API implemented under brayns/utils/image.
- ImageCodec interface to register image formats using ImageCodecRegistry.
- JPEG and PNG support using stb_image.
- Removed brayns IBL from the build.
- Removed some parts of the texture API to be able to compile.
- Removed perceptualdiff
- Re-render test images with stbi and update image validation process.
- Enabled warnings for building tests.

### CODE REVIEW

**Comment by NadirRoGue** [comment]
retest this please

**Comment by Adrien4193** [comment]
retest this please.

**Comment by Adrien4193** [comment]
retest this please

File: tests/helpers/ImageValidator.h [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                         const std::string &filename)
+    {
+        auto image = framebuffer.getImage();
+        return validate(image, filename);
+    }
+
+    static bool validate(const brayns::Image &image,
+                         const std::string &filename)
+    {
+        std::cout << "Validation of image '" << filename << "'.\n";
```
Comment (NadirRoGue) [inline tests/helpers/ImageValidator.h:56]: Are these cout messages needed for tests?

File: tests/helpers/ImageValidator.h [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        _saveIfNeeded(image, path);
+        auto reference = brayns::ImageDecoder::load(path);
+        return validate(image, reference);
+    }
+
+    static bool validate(const brayns::Image &image,
+                         const brayns::Image &reference)
+    {
+        auto evaluation = _evaluate(image, reference);
+        std::cout << "Evaluation result: " << evaluation << ".\n";
```
Comment (NadirRoGue) [inline tests/helpers/ImageValidator.h:67]: Are these cout messages needed for tests?

File: tests/helpers/ImageValidator.h (hunk 1 above)
Comment (Adrien4193) [reply tests/helpers/ImageValidator.h:56]: Yeah maybe not, there were some of them before and I replaced it by some debug info like the full file path of the template and the comparison between the evaluation and the threshold to have an idea of what is going on when a test fails.
You think it is better without ?

File: tests/helpers/ImageValidator.h (hunk 2 above)
Comment (Adrien4193) [reply tests/helpers/ImageValidator.h:67]: Yeah maybe not, there were some of them before and I replaced it by some debug info like the full file path of the template and the comparison between the evaluation and the threshold to have an idea of what is going on when a test fails.
You think it is better without ?

**Comment by Adrien4193** [comment]
retest this please

**Comment by Adrien4193** [comment]
retest this please

**Comment by Adrien4193** [comment]
retest this please

**Comment by Adrien4193** [comment]
retest this please

**Comment by Adrien4193** [comment]
retest this please

**Comment by NadirRoGue** [comment]
retest this please

**Comment by NadirRoGue** [comment]
retest this please

**Comment by NadirRoGue** [comment]
retest this please

**Review by NadirRoGue — APPROVED** [review]
LGTM

**Pull request merged**